lib/logstash/inputs/twitter.rb

```diff
@@ -104,8 +105,6 @@ class LogStash::Inputs::Twitter < LogStash::Inputs::Base
   config :rate_limit_reset_in, :validate => :number, :default => 300
 
   def register
-    require "twitter"
```

Comment:
IIRC the reason to require the library in `register` is to only "activate" the dependency when this plugin is registered, which reduces the footprint of a process that has a plugin installed but no pipeline using it.

Since `logstash-input-twitter` is _not_ bundled with any logstash distributions, I don't think this is terribly likely to occur, and since we are loading portions of the twitter library to monkey-patch them anyway, I am in favor of moving the whole require up as you have done.

Reply:
The purpose of this PR is to be able to move forward with https://github.com/elastic/logstash/pull/10934 where the twitter input will be included as a default plugin. 
But in any case, IIRC we also decided some time ago to move away from this idea of requiring dependant libraries in the `register` method in particular to fail early in the build process if there was any conflicts between dependant libraries.